library/crypto/sha512.c

```diff
@@ -321,6 +324,72 @@ sha384_stream (FILE *stream, void *resblock)
   return 0;
 }
 
+/* Compute SHA512 message digest for bytes read from file descriptor.
+   The resulting message digest number will be written into the 64
+   bytes beginning at RESBLOCK.  */
+int
+sha512_file_ctx (int fd, off_t offset, size_t count, struct sha512_ctx *ctx)
```

Comment:
No submodules - it brings horrible packaging and versioning problems, we already tried that and eventually moved every submodule into own package.
I used openssl first, but it has multithreading problems, so I put sha512 implementation into eblob from some FSF package, since it was only present in ubuntu and not in older rhel